5 - RPG/Code/RPG.CharacterBuilder/CharacterPropertyBase.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace RPG.CharacterBuilder
+{
+    public abstract class CharacterPropertyBase<T>
+    {
+        public virtual T Value { get; internal set; }
+        public abstract T DefaultValue { get; internal set; }
+        public virtual string invalidValueError { get; set; } = "Input Value is invalid";
+        public CharacterPropertyBase(T value)
+        {
+            if (IsValid(value))
+                Value = value;
+            else
+                Value = DefaultValue;
+        }
+        internal virtual OperationResult<CharacterPropertyBase<T>> SetValue(T value)
+        {
+            if (IsValid(value))
+            {
+                Value = value;
+                return new OperationResult<CharacterPropertyBase<T>>(this);
+            }
+            else
+            {
+                return new OperationResult<CharacterPropertyBase<T>>(invalidValueError);
+            }
+        }
+        public abstract bool IsValid(T value);
+    }
+}
```

Comment:
Overall I do like this base class, but it may be doing to much. You can have "layered" base classes that adds complexity so you don't force to much onto everything that uses it.